src/intersections/boxes.jl

```diff
@@ -34,3 +34,51 @@ function intersecttype(f::Function, b1::Box{Dim,T}, b2::Box{Dim,T}) where {Dim,T
     return FaceTouchingBoxes(Box(u, v)) |> f
   end
 end
+
+"""
+    intersecttype(function, ray, box)
+
+Calculate the intersection type of a `ray`` and a `box`` and apply function `f` to it.
+
+## References
+
+* Williams A, Barrus S, Morley R K, et al., 2005. [An efficient and robust ray-box
+  intersection algorithm]
+  (https://dl.acm.org/doi/abs/10.1145/1198555.1198748)
+"""
+function intersecttype(f::Function, r::Ray{3,T}, b::Box{3,T}) where {T}
+  invdir = one(T) ./ direction(r)
+  lo, up = coordinates.(extrema(b))
+  orig = coordinates(origin(r))
+
+  tmin = zero(T)
+  tmax = typemax(T)
+
+  for i in 1:3
```

Comment:
Ok, now that the algorithm is becoming clear, do we have anything special about Dim=3? Isn't this algorithm N-dimensional?

Reply:
@juliohm Yes. You are right. It is adapt for any N-dimensional.